src/repository/resources/ipres.rs

```diff
@@ -1709,12 +1709,128 @@ impl AddressFamily {
 }
 
 
+//------------ Ipv4Block -----------------------------------------------------
+
+/// Contains an IPv4 address block.
+///
+/// This type is a thin wrapper around [`IpBlock`] ensuring that this is an
+/// IPv4 address block.
+#[derive(Clone, Debug, Eq, PartialEq)]
+pub struct Ipv4Block(IpBlock);
+
+impl Ipv4Block {
+    /// Creates a new block covering all addresses.
+    pub fn all() -> Self {
+        Self(IpBlock::all())
+    }
+
+    /// Returns whether the block is prefix with address length zero.
+    pub fn is_slash_zero(&self) -> bool {
+        self.0.is_slash_zero()
+    }
+
+    /// The smallest address of the block.
+    pub fn min(&self) -> Ipv4Addr {
+        self.0.min().into()
+    }
+
+    /// The largest address of the block.
+    pub fn max(&self) -> Ipv4Addr {
+        self.0.max().into()
+    }
+}
+
+
+//--- From and FromStr
+
+impl From<Ipv4Block> for IpBlock {
+    fn from(src: Ipv4Block) -> IpBlock {
+        src.0
+    }
+}
+
+impl str::FromStr for Ipv4Block {
+    type Err = FromStrError;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        Ok(Self(IpBlock::from_v4_str(s)?))
+    }
+}
+
+
+//--- Display
```

Comment:
As `IpBlock` has `FromIterator<IpBlock>` should there also be `FromIterator<Ipv4Block>` here?